### PR TITLE
feat: change generic color class names

### PR DESCRIPTION
**Pull request by nekosaur**
## Description

```
// v2
.indigo
.indigo--text

.indigo.lighten-2
.indigo--text.text--darken-1

// v3
.bg-indigo
.text-indigo

.bg-indigo-lighten-2
.text-indigo-darken-1
```

## Motivation and Context

This aligns generic color classes with those generated from the theme system, which means that the generic ones can be used in `color` prop without any additional work.

This is a breaking change.

## Markup:

<details>

```vue
// Paste your FULL Playground.vue here
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/docs/src/examples/color/text-classes.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
   <div>
-    Lorem ipsum dolor sit amet, <strong class="red--text text--lighten-1">inciderint</strong> definitionem est ea, explicari prodesset eam id. Mazim doctus vix an. <span class="indigo--text text--darken-2">Amet causae probatus nec ex</span>.
+    Lorem ipsum dolor sit amet, <strong class="text-red-lighten-1">inciderint</strong> definitionem est ea, explicari prodesset eam id. Mazim doctus vix an. <span class="text-indigo--darken-2">Amet causae probatus nec ex</span>.
```
Comment (Mert75) [inline packages/docs/src/examples/color/text-classes.vue:3]: `class="text-indigo--darken-2"` uses two dashes, this should be only one

File: packages/docs/src/pages/en/styles/colors.md [hunk 2]
```diff
@@ -31,21 +31,21 @@ Vuetify has an optional javascript color pack that you can import and use within
...
     themes: {
       light: {
-        primary: colors.red.darken1, // #E53935
-        secondary: colors.red.lighten4, // #FFCDD2
-        accent: colors.indigo.base, // #3F51B5
+        dark: false,
+        colors: {
+          primary: colors.red.darken1, // #E53935
+          secondary: colors.red.lighten4, // #FFCDD2
+          accent: colors.indigo.base, // #3F51B5
```
Comment (johnleider) [inline packages/docs/src/pages/en/styles/colors.md:46]: do we still use accent?

**Review by johnleider — APPROVED** [review]
Other than my one question, looks good to me.

**Review by johnleider — COMMENTED** [review]
It looks good. What is our stance on the global .v-application class? I believe we talked about removing it.

**Comment by nekosaur** [comment]
I think it would make sense to make them global, since that's how the theme classes work

**Comment by johnleider** [comment]
> I think it would make sense to make them global, since that's how the theme classes work

@KaelWD could you chime in?

**Comment by KaelWD** [comment]
Everything should be global in v3

**Pull request merged**